pkgs/build-support/dotnet/make-nuget-deps/default.nix

```diff
@@ -1,23 +1,33 @@
-{ symlinkJoin
-, fetchurl
-, stdenvNoCC
-, lib
-, unzip
-, patchNupkgs
-, nugetPackageHook
-, fetchNupkg
+{
+  symlinkJoin,
+  lib,
+  fetchNupkg,
 }:
-lib.makeOverridable(
-  { name
-  , nugetDeps ? import sourceFile
-  , sourceFile ? null
-  , installable ? false
+lib.makeOverridable (
+  {
+    name,
+    nugetDeps ? null,
+    sourceFile ? null,
+    installable ? false,
   }:
   (symlinkJoin {
     name = "${name}-nuget-deps";
-    paths = nugetDeps {
-      fetchNuGet = args: fetchNupkg (args // { inherit installable; });
-    };
-  }) // {
-    inherit sourceFile;
+    paths =
+      let
+        loadDeps =
+          if nugetDeps != null then
+            nugetDeps
+          else if lib.hasSuffix ".nix" sourceFile then
+            assert (lib.isPath sourceFile);
+            import sourceFile
+          else
+            { fetchNuGet }: builtins.map fetchNuGet (lib.importJSON sourceFile);
```

Comment:
Shouldn't the assert be in the `else` branch, not the `else if`? Or perhaps both branches?

Reply:
I agree it's not really needed. TBH I have a feeling I may have been the one to add it originally?

I would see more value in the assert if it was using `lib.assertMsg` to improve the error message that I believe would happen _anyway_.

If not for https://github.com/NixOS/nixpkgs/pull/362278#discussion_r1873302507 I would suggest dropping the assert. cc @Atemu

Reply:
No, the assert was there previously:
https://github.com/NixOS/nixpkgs/blob/275baaeee3b73087ef3de602c583e0b48399f551/pkgs/build-support/dotnet/add-nuget-deps/default.nix#L29-L30

So we should keep it to avoid any unwanted behavior changes.

Reply:
Ah, traced it back to cc9c59c4780a097ef7b988daeb118269e73d4730